src/view.c

```diff
@@ -19,6 +20,33 @@
 #define LAB_FALLBACK_WIDTH  640
 #define LAB_FALLBACK_HEIGHT 480
 
+void
+view_query_free(struct view_query *query)
+{
+	wl_list_remove(&query->link);
+	free(query->identifier);
+	free(query->title);
+	free(query);
+}
+
+bool
+view_matches_query(struct view *view, struct view_query *query)
+{
+	bool match = true;
+
+	const char *identifier = view_get_string_prop(view, "app_id");
+	if (match && query->identifier && identifier) {
+		match &= match_glob(query->identifier, identifier);
+	}
+
+	const char *title = view_get_string_prop(view, "title");
+	if (match && query->title && title) {
+		match &= match_glob(query->title, title);
+	}
+
+	return match;
+}
```

Comment:
2nd point already discussed via IRC, comment here just to ensure we don't forget about it: empty queries (caused by for example the wrong arguments) should not match, missing queries should (at least for `ForEach`).

So this function should return `false` when none of the `query` arguments are set.
The missing query part is already handled correctly in `run_if_actions()`

Reply:
Fixed.

Reply:
This is ugly though :( It would be better not to create such query in the first place.